microsoft-edge/progressive-web-apps-chromium/how-to/best-practices.md

```diff
@@ -0,0 +1,205 @@
+* Non-trivial computing tasks with [WebAssembly](https://developer.mozilla.org/docs/WebAssembly).
+* Reading and writing to files with the [File System Access API](https://developer.mozilla.org/docs/Web/API/File_System_Access_API).
+
+Ensure your app's most important tasks can be done across all browsers and devices. See [Test on multiple browsers and devices](#test-on-multiple-browsers-and-devices).
+
+
+<!-- ====================================================================== -->
+## Make your app look and feel like a real app
+
+Users install apps on their devices to achieve specific tasks, and they have certain expectations about how these tasks are presented in the user interface and how they integrate in the host operating system.
+
+To let your users achieve your app's most important tasks easily and in a familiar way, make design choices such as:
+
+* Don't use a big header area like websites do for navigation to other pages. Use a menu metaphor instead.
+* Don't use a big footer area like websites do for more links and information.
+* Use the `system-ui` font to make your content feel more native and load faster.
+* Use less links, and prefer bigger hit targets.
+* Support the operating system's light and dark themes by using the [prefers-color-scheme](https://developer.mozilla.org/docs/Web/CSS/@media/prefers-color-scheme) CSS media feature.
+* Provide a `background-color` and `theme-color` in your [web app manifest](https://developer.mozilla.org/docs/Web/Manifest) to customize the application window.
+* Focus on the most important tasks, de-clutter de content, and streamline the user interface.
+* Re-organize the user interface based on the device's form factor, by using CSS [grid](https://developer.mozilla.org/docs/Web/CSS/CSS_Grid_Layout), [Flexbox](https://developer.mozilla.org/docs/Web/CSS/CSS_Flexible_Box_Layout), [Media queries](https://developer.mozilla.org/docs/Web/CSS/Media_Queries), and [Responsive images](https://developer.mozilla.org/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images).
+* Improve perceived performance by registering user interactions immediately.
+* Consider using [skeleton screens](https://css-tricks.com/building-skeleton-screens-css-custom-properties/) if rendering the next screen takes time.
+
+
+<!-- ====================================================================== -->
+## Test on multiple browsers and devices
+
+With Progressive Web Apps, you deliver an app for all devices, from one codebase. To make sure your app works everywhere, test it on multiple browsers, operating systems, and devices.
+
+Make sure the most important scenarios of your app work everywhere, and progressively enhance the experience on supporting devices.
+
+To test your app in multiple environment, consider the following resources:
+
+* [Cross-browser testing](https://developer.mozilla.org/docs/Learn/Tools_and_testing/Cross_browser_testing) on MDN
```

Comment:
```suggestion
* [Cross-browser testing](https://developer.mozilla.org/docs/Learn/Tools_and_testing/Cross_browser_testing) on MDN.
```

Reply:
Good point, I'll remove the "on MDN" part.